distributed_nhc/export_nhc_result_to_kusto.py

```diff
@@ -60,13 +63,155 @@ def ingest_debug_log(debug_file, creds, ingest_url, database, debug_table_name):
+    n = "CPU" if stream_Copy_str else "GPU"
+
+    if n == "CPU":
+        ib_write_lb_mlx5_ib_cmd = f"cat {results_file} | grep -o 'ib_write_lb_mlx5_ib[0-7]: .*'"
+        ib_write_lb_mlx5_ib_str = run_command(ib_write_lb_mlx5_ib_cmd)
+        ib_write_lb_mlx5_ib_str = sorted(ib_write_lb_mlx5_ib_str.strip().split("\n"), key=natural_sort_key)
+        ib_write_lb_mlx5_ib_str = '\n'.join(ib_write_lb_mlx5_ib_str) # convert to string
+
+        # stream_Copy_str gotten above
+
+        stream_Add_cmd = f"cat {results_file} | grep -o 'stream_Add: .*'"
+        stream_Add_str = run_command(stream_Add_cmd)
+
+        stream_Scale_cmd = f"cat {results_file} | grep -o 'stream_Scale: .*'"
+        stream_Scale_str = run_command(stream_Scale_cmd)
+
+        stream_Triad_cmd = f"cat {results_file} | grep -o 'stream_Triad: .*'"
+        stream_Triad_str = run_command(stream_Triad_cmd)
+
+        data_string = "\n".join([ib_write_lb_mlx5_ib_str, stream_Copy_str, stream_Add_str, stream_Scale_str, stream_Triad_str])
+
+        result = {"IB_WRITE_NON_GDR": {}, "stream_Copy": {}, "stream_Add": {}, "stream_Scale": {}, "stream_Triad": {}}
+
+        # Split the string by lines and create key-value pairs
+        for line in data_string.strip().split("\n"):
+            key, value = line.split(":")
+            if key.startswith("ib_write_lb_mlx5_ib"):
+                result["IB_WRITE_NON_GDR"][key] = str(value.strip())
+            elif key.startswith("stream_Copy"):
+                result["stream_Copy"]= str(value.strip())
+            elif key.startswith("stream_Add"):
+                result["stream_Add"]= str(value.strip())
+            elif key.startswith("stream_Scale"):
+                result["stream_Scale"]= str(value.strip())
+            elif key.startswith("stream_Triad"):
+                result["stream_Triad"]= str(value.strip())
+
+    elif n == "GPU":
+        ib_write_lb_mlx5_ib_cmd = f"cat {results_file} | grep -o 'ib_write_lb_mlx5_ib[0-7]: .*'"
+        ib_write_lb_mlx5_ib_str = run_command(ib_write_lb_mlx5_ib_cmd)
+        ib_write_lb_mlx5_ib_str = sorted(str(ib_write_lb_mlx5_ib_str).strip().split("\n"), key=natural_sort_key)
+
+        H2D_GPU_cmd = f"cat {results_file} | grep -o 'H2D_GPU_[0-7]: .*'"
+        H2D_GPU_str = run_command(H2D_GPU_cmd)
+
+        D2H_GPU_cmd = f"cat {results_file} | grep -o 'D2H_GPU_[0-7]: .*'"
+        D2H_GPU_str = run_command(D2H_GPU_cmd)
+
+        P2P_GPU_cmd = f"cat {results_file} | grep -o 'P2P_GPU_[0-7]_[0-7]: .*'"
+        P2P_GPU_str = run_command(P2P_GPU_cmd)
+
+        nccl_all_red_cmd = f"cat {results_file} | grep -o 'nccl_all_red: .*'"
+        nccl_all_red_str = run_command(nccl_all_red_cmd)
+
+        nccl_all_red_lb_cmd = f"cat {results_file} | grep -o 'nccl_all_red_lb: .*'"
+        nccl_all_red_lb_str = run_command(nccl_all_red_lb_cmd)
+
+        data_string = "\n".join([ib_write_lb_mlx5_ib_str, H2D_GPU_str, D2H_GPU_str, P2P_GPU_str, nccl_all_red_str, nccl_all_red_lb_str])
```

Comment:
It looks like I am failing here. The issue is that "ib_write_lb_mlx5_ib_str" seems to be a list. In the CPU section we converted it to a string. We need to do the same for the GPU section. i.e.  ib_write_lb_mlx5_ib_str = '\n'.join(ib_write_lb_mlx5_ib_str) # convert to string